server/balancer_test.go

```diff
@@ -116,16 +112,74 @@ func (c *testClusterInfo) updateSnapshotCount(storeID uint64, snapshotCount int)
 	c.putStore(store)
 }
 
+func (c *testClusterInfo) updateStorageRatio(storeID uint64, storageRatio float64) {
+	store := c.getStore(storeID)
+	store.stats.Capacity = uint64(1024)
+	store.stats.Available = uint64(float64(store.stats.Capacity) * (1 - storageRatio))
+	c.putStore(store)
+}
+
 func newTestScheduleConfig() (*ScheduleConfig, *scheduleOption) {
 	cfg := NewConfig()
 	cfg.adjust()
-	cfg.Schedule.MinLeaderCount = 1
-	cfg.Schedule.MinRegionCount = 1
 	cfg.Schedule.ScheduleInterval.Duration = time.Nanosecond
 	opt := newScheduleOption(cfg)
 	return &cfg.Schedule, opt
 }
 
+var _ = Suite(&testBalanceSpeedSuite{})
+
+type testBalanceSpeedSuite struct{}
+
+type testBalanceSpeedCase struct {
+	sourceCount    uint64
+	targetCount    uint64
+	balanceLimit   uint64
+	expectedResult bool
+}
+
+func (s *testBalanceSpeedSuite) TestBalanceSpeed(c *C) {
+	testCases := []testBalanceSpeedCase{
+		// diff > 1
+		{1, 0, 0, false},
+		{2, 0, 0, true},
+		{4, 0, 1, true},
+		// diff > 2
+		{10, 0, 2, true},
+		{10, 6, 1, true},
+		{10, 7, 0, true},
+		{10, 9, 0, false},
+		// diff > 4
+		{100, 0, 25, true},
+		{100, 95, 1, true},
+		{100, 97, 0, false},
+		// diff > 8
+		{1000, 0, 250, true},
+		{1000, 991, 2, true},
+		{1000, 993, 1, false},
+		// diff > 16
+		{10000, 0, 2500, true},
+		{10000, 9983, 4, true},
+		{10000, 9985, 3, false},
+	}
+
+	s.testBalanceSpeed(c, testCases, 1)
+	s.testBalanceSpeed(c, testCases, 10)
+	s.testBalanceSpeed(c, testCases, 100)
+	s.testBalanceSpeed(c, testCases, 1000)
+}
+
+func (s *testBalanceSpeedSuite) testBalanceSpeed(c *C, tests []testBalanceSpeedCase, capaGB uint64) {
+	for _, t := range tests {
+		sourceCount := t.sourceCount
+		sourceRatio := float64(t.sourceCount) / float64(capaGB)
```

Comment:
why ratio is calculated by sourceCount / capaGB? 

capaGB is for capacity GB, it seems a size, not count.